programmer/tinyprog/__init__.py

```diff
@@ -161,7 +161,7 @@ def _read_metadata(self):
         import math
         meta_roots = (
             [self._parse_json(self.prog.read_security_register_page(p).replace(b"\x00", b"").replace(b"\xff", b"")) for p in [1, 2, 3]] +
-            [self._parse_json(self.prog.read(int(math.pow(2, p) - (4 * 1024)), (4 * 1024)).replace(b"\x00", b"").replace(b"\xff", b"")) for p in [17, 18, 19, 20, 21, 22, 23, 24]]
+            [self._parse_json(self.prog.read(int(math.pow(2, p) - (4 * 1024)), (4 * 1024) - 256).replace(b"\x00", b"").replace(b"\xff", b"")) for p in [17, 18, 19, 20, 21, 22, 23, 24]]
```

Comment:
Why did you change this to only read 256 bytes from the latest 4KB sector? The original code reads the full 4KB in order to give more space for the JSON metadata.

Reply:
When I initialize flash using scripts in "initialize" directory and if
I have jump command which must start at last 256 bytes, then it
makes no sense to concatenate this to JSON string.

additional 256 binary of jump command concatenated to JSON
makes my python exit with error :)

Reply:
Ok, so it is keeping 4KB minus the 256 bytes required for the jump code.  Now that makes sense to me.

Reply:
Yes, yes, golden jump code must be at last 256 bytes, its location is hardcoded by lattice